flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapManager.java

```diff
@@ -1201,23 +1204,24 @@ private record FieldPosition(Map<Path, List<TextEdit>> textEdits, LinePosition p
 
     }
 
-    private List<IntermediateClause> getQueryIntermediateClause(QueryPipelineNode queryPipelineNode) {
-        List<IntermediateClause> intermediateClauses = new ArrayList<>();
+    private List<Clause> getQueryIntermediateClause(QueryPipelineNode queryPipelineNode) {
+        List<Clause> intermediateClauses = new ArrayList<>();
         for (IntermediateClauseNode intermediateClause : queryPipelineNode.intermediateClauses()) {
             SyntaxKind kind = intermediateClause.kind();
             switch (kind) {
                 case FROM_CLAUSE -> {
                     FromClauseNode fromClauseNode = (FromClauseNode) intermediateClause;
                     TypedBindingPatternNode typedBindingPattern = fromClauseNode.typedBindingPattern();
-                    FromClause fromClause = new FromClause(typedBindingPattern.typeDescriptor().toSourceCode().trim(),
-                            typedBindingPattern.bindingPattern().toSourceCode().trim(),
-                            fromClauseNode.expression().toSourceCode().trim());
-                    intermediateClauses.add(new IntermediateClause("from", fromClause));
+                    intermediateClauses.add(new Clause("from",
```

Comment:
I would extract `"from"`, `"select"`, and `"collect"` values to constants

Reply:
Fixed